R/ScpModel-Class.R

```diff
@@ -308,6 +308,7 @@ scpModelDf <- function(object, name, filtered = TRUE) {
     out <- scpModelFitElement(
         object, name, "Df", filtered, .runWorkflowMessage
     )
+    out <- lapply(out, function(x) if (length(x) == 0) NA else x)
```

Comment:
# Issue:

When certain degrees of freedom (Df) could not be estimated, they were assigned a value of numeric(0). However, when using unlist, these numeric(0) values were inadvertently dropped, resulting in a returned Df vector that was identical whether or not filtering was applied (i.e., filtering = TRUE). This caused an issue during the computation of the vector p (n-df), as the vectors n and Df no longer had matching lengths.
# Solution:

To address this, the numeric(0) values were replaced with NA, preserving the length of the Df vector. This ensures that Df remains aligned with n, even if some degrees of freedom were not estimated.

However, the introduction of NA values in the Df vector naturally leads to NA values in the computed p vector. To avoid this issue, these NA values in p are subsequently converted to 0 within the scpModelP getter.

Reply:
This is an excellent solution!